data/countries/JP.yaml

```diff
@@ -25,90 +30,283 @@ holidays:
+        active:
+          - from: 2007-01-01
+      05-05:
+        name:
+          en: Children's Day
+          jp: こどもの日
+      substitutes 05-05 if sunday then next monday:
         substitute: true
         name:
           en: Children's Day
           jp: こどもの日
+        active:
+          - from: 1973-04-12
+      07-20:
+        substitute: true
+        name:
+          en: Marine Day
+          jp: 海の日
+        active:
+          - from: 1996-01-01
+            to: 2002-12-31
       3rd monday in July:
         name:
           en: Marine Day
           jp: 海の日
+        disable:
+          - '2020-07-20'
+        enable:
+          - '2020-07-23'
+        active:
+          - from: 2003-01-01
       08-11 and if sunday then next monday:
         substitute: true
         name:
           en: Mountain Day
           jp: 山の日
+        disable:
+          - '2020-08-11'
+        enable:
+          - '2020-08-10'
+        active:
+          - from: 2016-01-01
+      09-15:
+        name:
+          en: Respect-for-the-Aged Day
+          jp: 敬老の日
+        active:
+          - from: 1966-01-01
+            to: 2002-12-31
+      substitutes 09-15 and if sunday then next monday:
+        substitute: true
+        name:
+          en: Respect-for-the-Aged Day
+          jp: 敬老の日
+        active:
+          - from: 1973-04-12
+            to: 2002-12-31
       3rd monday in September:
         name:
           en: Respect-for-the-Aged Day
           jp: 敬老の日
+        active:
+          - from: 2003-01-01
       09-22 if 09-21 and 09-23 is public holiday:
         name:
-          en: Public Holiday
-          jp: 祝日
+          en: Citizens' Holiday
+          jp: 国民の休日
       september equinox in +09:00:
         name:
           en: Autumnal Equinox Day
           jp: 秋分の日
-      substitutes 09-23 if sunday then next monday:
+        active:
+          - from: 1948-07-20
+      substitutes september equinox in +09:00 and if sunday then next monday:
         substitute: true
         name:
           en: Autumnal Equinox Day
           jp: 秋分の日
-      2nd monday in October:
+        active:
+          - from: 1973-04-12
+      10-10:
         name:
           en: Sports Day
           jp: 体育の日
-      11-03 and if sunday then next monday:
+        active:
+          - from: 1966-01-01
+            to: 1999-12-31
+      substitutes 10-10 and if sunday then next monday:
+        substitute: true
+        name:
+          en: Sports Day
+          jp: 体育の日
+        active:
+          - from: 1973-04-12
+            to: 1999-12-31
+      '2nd monday in October #1':
+        name:
+          en: Sports Day
+          jp: 体育の日
+        active:
+          - from: 2000-01-01
+            to: 2019-12-31
+      '2nd monday in October #2':
+        name:
+          en: Sports Day
+          jp: スポーツの日
+        disable:
+          - '2020-10-12'
+        enable:
+          - '2020-07-24'
+        active:
+          - from: 2020-01-01
+      '2019-10-22':
```

Comment:
I do not think that the single quotes are needed in this YAML key.

Reply:
This quote is necessary I think. 

See: https://github.com/commenthol/date-holidays/blob/master/docs/specification.md#fixed-date

Reply:
You are correct, sorry.